### PR TITLE
Add LLM for ChatGLM(2)-6B API

### PR DESCRIPTION
**Pull request by mlot**
Description:
Add LLM for ChatGLM-6B & ChatGLM2-6B API

Related Issue: 
Will the langchain support ChatGLM? #4766
Add support for selfhost models like ChatGLM or transformer models #1780

Dependencies: 
No extra library install required. 
It wraps api call to a ChatGLM(2)-6B server(start with api.py),  so api endpoint is required to run.

Tag maintainer:  @mlot 

Any comments on this PR would be appreciated.

### CODE REVIEW

**Comment by mlot** [comment]
ChatGLM is a quite(if not the most) popular open bilingual LLM in China, which has over 4.2M downloads and 32k github stars, having upstream support from langchain will be great. @hwchase17, thanks for your time to review.

**Comment by hetaoBackend** [comment]
It's cool to see the ChatGLM model could be integrated into Langchain! Awesome! @mlot

File: langchain/llms/chatglm.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+            )
+    """
+    
+    endpoint_url: str = (
+        "http://127.0.0.1:8000/"
+    )
+    """Endpoint URL to use."""
+    model_kwargs: Optional[dict] = None
+    """Key word arguments to pass to the model."""
+    max_token: int = 20000
```
Comment (hetaoBackend) [inline langchain/llms/chatglm.py:28]: Is it possible to support `top_p` in the parameters to get more flexibility?

Comment (mlot) [reply langchain/llms/chatglm.py:28]: thanks for pointing it out, added.

**Comment by baskaryan** [comment]
thanks @mlot!

**Comment by mlot** [comment]
thanks @baskaryan !

**Comment by zhanaali** [comment]
Does llms.chatglm support stream output?

**Pull request merged**